lib/diaspora/federation/entities.rb

```diff
@@ -200,37 +177,29 @@ def self.reshare(reshare)
         )
       end
 
-      def self.retraction(target)
+      def self.retraction(retraction)
+        case retraction.data[:target_type]
+        when "Contact"
```

Comment:
I also thought that the `Retraction` class actually has two different responsibilities: first is submitting a job to sidekiq and second is passing data from sidekiq worker to entity constructor. Actually this responsibilities aren't too bound IMO, so I think they could actually be separated to different objects. Maybe you don't want to make that big refactorings, but if you want, I can try to make some code changes to show my idea.